### PR TITLE
frontend: Add a warning filter  switch to events table

### PR DESCRIPTION
**Pull request by ashu8912**
Users want to filter simply by clicking a switch to list all the warning type events, this patch adds that to the events table.

### CODE REVIEW

**Comment by ashu8912** [comment]
Screenshot 2023-02-06 at 11 38 38 AM
Screenshot 2023-02-06 at 11 38 48 AM

**Comment by ashu8912** [comment]
https://user-images.githubusercontent.com/30126128/216931390-18f1cbe3-c20d-4ca4-8dd0-8a5190e71940.mp4

**Comment by joaquimrocha** [comment]
The mock ups were actually not approved yet so this will need some changes as the current UX is not optimal.

**Comment by joaquimrocha** [comment]
Here is the updated mockup. Also, it shouldn't hide the switch when the filter is opened (the filter should work together with the warning switch).
Mockup showing the warning switch aligned to the left

**Comment by ashu8912** [comment]
@joaquimrocha updated this with the new designs

File: frontend/src/components/common/SimpleTable.tsx [hunk 1]
```diff
@@ -153,6 +154,7 @@ export default function SimpleTable(props: SimpleTableProps) {
     columns,
     data,
     filterFunction = null,
+    actionFilterFunction = null,
```
Comment (joaquimrocha) [inline frontend/src/components/common/SimpleTable.tsx:157]: I'd rather keep a single filter function. This seems too intimate to this PR's use-case and it's not clear what exactly `actionFilterFunction` does or why it's needed.

File: frontend/src/components/common/SectionHeader.tsx [hunk 2]
```diff
@@ -28,6 +29,7 @@ export interface SectionHeaderProps {
   actions?: React.ReactNode[] | null;
   noPadding?: boolean;
   headerStyle?: HeaderStyle;
+  actionsBesideTitle?: React.ReactNode[];
```
Comment (joaquimrocha) [inline frontend/src/components/common/SectionHeader.tsx:32]: Let's call this `titleSideActions` please.

**Review by joaquimrocha — CHANGES_REQUESTED** [review]
It's almost there. Left new comments.

File: frontend/src/components/common/SimpleTable.tsx (hunk 1 above)
Comment (ashu8912) [reply frontend/src/components/common/SimpleTable.tsx:157]: there is ```filterFunc``` which uses the search capabilities from store using hooks like ```useFilterFunc()```  but then we need another filterFunction which works beside the normal filtering, that's why this new prop

File: frontend/src/components/common/SectionHeader.tsx (hunk 2 above)
Comment (ashu8912) [reply frontend/src/components/common/SectionHeader.tsx]: updated

**Review by joaquimrocha — APPROVED** [review]

File: frontend/src/components/common/SimpleTable.tsx (hunk 1 above)
Comment (joaquimrocha) [reply frontend/src/components/common/SimpleTable.tsx:157]: I forgot to insist on getting a solution for this. I will try to fix it in another PR.

Comment (joaquimrocha) [reply frontend/src/components/common/SimpleTable.tsx:157]: @ashu8912 I have added the fix in #979 . Please take a look.

**Pull request merged**